### PR TITLE
Timur/measurement

### PR DESCRIPTION
**Pull request by ttoktassynov**
### Purpose
This pull request adds a measurement node capability and also changes the way classical registers are named upon construction in their javascript files.

### Changes

I have implemented measurement node, which connects to a quantum register output and in the editor allows user to choose a classical bit index of the classical register to save measurements to.

To achieve that I have modified the classical-register.js file to setup a HttpAdmin get request # in order to communicate data about number of classical bits to the measurement node. Also I setup a HttpAdmin post request handler for the classical register to persist value of classical bits number on the frontend. See this a38e7b8bef820f4b25700455881dd9c2e7e7d495 for reference
 
Finally, I have slightly changed the naming convention of the classical registers. Because I don't have access to global context's structure array, thus I will not know the registers names until after the flow has started. That is why I changed the naming of the classical register variables in the script to be: 
```
this.registerVar = 'cr' + node.id.replace('.', '_');
```

### CODE REVIEW

**Review by XxWirusonxX — APPROVED** [review]
looks fine too me

File: quantum/nodes/classical-register/classical-register.js [hunk 1]
```diff
@@ -1,18 +1,20 @@
 module.exports = function(RED) {
   'use strict';
-
+  let classicalRegister = null;
   function ClassicalRegisterNode(config) {
     // Creating node with properties and context
+    classicalRegister = this;
```
Comment (Theo-Reignier) [inline quantum/nodes/classical-register/classical-register.js:6]: Very happy with everything else but here you could use the node variable defined a few lines below which correspond to the same object as classicalRegister here

**Review by Theo-Reignier — DISMISSED** [review]
Very happy with this first version of the measurement node.

I just left a comment because I believe you created a redundant variable in classical-register.js

**Comment by Theo-Reignier** [comment]
I resolved the conflicts with Classical register in order to merge but it removed the 2 reviews....

**Comment by louislefevre** [comment]
Will review this later today, sorry for the delay

**Review by ttoktassynov — COMMENTED** [review]
Thanks for reviewing guys, can anybody finish reviewing, so it can be merged?

**Review by Theo-Reignier — DISMISSED** [review]
All Good

**Comment by louislefevre** [comment]
LGTM, I just added a commit to lint the HTML file. Currently the `npm run lint` command doesn't work on HTML files and I'm using my own (separate) plugin to do it. I'll try and improve the command to include HTML files in the future.

**Review by louislefevre — APPROVED** [review]

**Pull request merged**